adapter/ipc/adapter_ipc_connection.cpp

```
@@ -32,17 +32,60 @@ void InvokeContained(const Callback &callback, Args &&...args) {
 
 AdapterIpcConnection::AdapterIpcConnection(
     IAdapterIpcSocket &socket, const IIpcFrameCodec &codec,
-    AdapterIpcConnectionCallbacks callbacks)
-    : socket_(socket), codec_(codec), callbacks_(std::move(callbacks)) {}
+    AdapterIpcConnectionCallbacks callbacks,
+    std::chrono::milliseconds establishmentTimeout, ClockNow clockNow)
+    : socket_(socket), codec_(codec), callbacks_(std::move(callbacks)),
+      establishmentTimeout_(establishmentTimeout),
+      clockNow_(std::move(clockNow)) {}
 
 AdapterIpcConnection::~AdapterIpcConnection() { Stop(); }
```

Cppcheck (2.21.0)

[error] Unhandled exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)